src/models/db/case.js

```diff
@@ -1,72 +1,115 @@
-      if (!Array.isArray(data.notaries)) {
-        data.notaries = [ data.notaries ]
+function transaction (callback) {
+  return knex.transaction(co.wrap(callback))
+}
+
+function * insertCases (caseObject, options) {
+  const persistentCaseObjects = _.map(caseObject, convertToPersistent)
+  const myKnex = getTransaction(knex, options)
+  if (persistentCaseObjects.length > 0) {
+    return myKnex(TABLE_NAME).insert(persistentCaseObjects).then()
+  }
+}
+
+function insertCase (caseObject, options) {
+  return getTransaction(knex, options)(TABLE_NAME).insert(convertToPersistent(caseObject)).then()
+}
+
+function updateCase (caseObject, options) {
+  return getTransaction(knex, options)(TABLE_NAME).update(convertToPersistent(caseObject))
+    .where('CASE_ID', caseObject.primary_key)
+    .then()
+}
+
+function getByKey (key, value, options) {
+  return getTransaction(knex, options)
+    .from(TABLE_NAME)
+    .select()
+    .where(key, value)
+    .then((results) => results.map(convertFromPersistent))
+    .then((results) => {
+      if (results.length === 1) {
+        return results[0]
+      } else if (results.length === 0) {
+        return null
+      } else {
+        assert(false,
+          results.length + ' cases with ' + key + ':' + value)
       }
-      return data
-    }
+    })
+}
 
-    static convertToPersistent (data) {
-      data.execution_condition = JSON.stringify(data.execution_condition)
-      data.exec_cond_fulfillment = JSON.stringify(data.exec_cond_fulfillment)
-      data.notification_targets = JSON.stringify(data.notification_targets)
-      data.notaries = data.notaries[0]
-      return data
-    }
+function getCaseByPrimaryKey (primaryKey, options) {
+  return getByKey('CASE_ID', primaryKey, options)
+}
 
-    isFinalized () {
-      return this.state === 'executed' || this.state === 'rejected'
-    }
-  }
+function getCaseByUuid (caseUuid, options) {
+  return getByKey('CASE_UUID', caseUuid, options)
+}
 
-  Case.validateExternal = validator.create('Case')
-  Case.tableName = 'cases'
+function convertFromPersistent (data) {
+  const caseObject = {
+    primary_key: data.CASE_ID,
+    id: data.CASE_UUID,
+    execution_condition: data.EXECUTION_CONDITION_DIGEST,
+    expires_at: data.EXPIRES_DTTM,
+    notification_targets: JSON.parse(data.NOTIFICATION_TARGETS),
+    state: CASE_STATUS_IDS[data.CASE_STATUS_ID]
+  }
+  if (!Array.isArray(data.NOTARIES)) {
+    caseObject.notaries = [ data.NOTARIES ]
+  }
+  if (data.EXEC_COND_FULFILLMENT) {
+    caseObject.exec_cond_fulfillment = data.EXEC_COND_FULFILLMENT
+  }
+  data = caseObject
+  return data
+}
 
-  PersistentKnexModelMixin(Case, knex)
+function convertToPersistent (data) {
+  const persistentCase = {
+    CASE_UUID: data.id,
+    CASE_STATUS_ID: CASE_DESCRIPTIONS[data.state],
+    EXECUTION_CONDITION_DIGEST: data.execution_condition,
+    EXEC_COND_FULFILLMENT: data.exec_cond_fulfillment,
+    EXPIRES_DTTM: data.expires_at,
+    NOTARIES: data.notaries[0],
```

Comment:
We pick the first notary in the array to persist in the DB, and then convert back to an array when we read from the DB. I'm inclined to leave as is for now. This code has been in here for a while, and it's not related to this PR, so I'd prefer to address any cleanup in a separate task.